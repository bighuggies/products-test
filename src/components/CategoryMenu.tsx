import { inject, observer } from 'mobx-react';
import * as React from 'react';
import { NavLink, withRouter } from 'react-router-dom';
import styled from 'styled-components';

import { CategoryStore } from '../stores/CategoryStore';

const StyledMenu = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;

  > li + li {
    margin-left: 1.5rem;
  }
`;

const StyledMenuLink = styled(NavLink)`
  text-decoration: none;
  color: initial;
  font-size: 1.25rem;

  &.is-active {
    font-weight: 600;
    text-decoration: underline;
  }
`;

class CategoryMenu extends React.Component<{ categoryStore?: CategoryStore }> {
  public render() {
    return (
      <nav aria-label="Categories">
        <StyledMenu>
          {this.props.categoryStore &&
            this.props.categoryStore.activeCategories.map(c => (
              <li key={c.id}>
                <StyledMenuLink activeClassName="is-active" to={`/${c.id}`}>
                  {c.title}
                </StyledMenuLink>
              </li>
            ))}
        </StyledMenu>
      </nav>
    );
  }
}

export default inject(CategoryStore.STORE_NAME)(
  withRouter(observer(CategoryMenu) as any)
);
