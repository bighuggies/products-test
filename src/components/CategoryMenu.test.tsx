import { mount } from 'enzyme';
import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';

import CategoryMenu from './CategoryMenu';

test('renders without crashing', () => {
  const mounted = mount(
    <MemoryRouter keyLength={0}>
      <CategoryMenu categories={[{ id: '1', title: '1', hidden: false }]} />
    </MemoryRouter>
  );

  expect(mounted.debug()).toMatchSnapshot();
});
