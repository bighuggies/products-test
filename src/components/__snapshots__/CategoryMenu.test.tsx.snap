// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
"<MemoryRouter keyLength={0}>
  <Router history={{...}}>
    <withRouter(CategoryMenu) categories={{...}}>
      <Route>
        <CategoryMenu categories={{...}} match={{...}} location={{...}} history={{...}} staticContext={[undefined]}>
          <nav aria-label=\\"Categories\\">
            <Styled(styled.ul)>
              <styled.ul className=\\"sc-bwzfXH gtuzxh\\" innerRef={[undefined]}>
                <ul className=\\"sc-bwzfXH gtuzxh sc-bdVaJa hTzUzq\\">
                  <li>
                    <Styled(NavLink) activeClassName=\\"is-active\\" to=\\"/1\\">
                      <NavLink className=\\"sc-htpNat jgryAF\\" activeClassName=\\"is-active\\" to=\\"/1\\" aria-current=\\"page\\">
                        <Route path=\\"\\\\\\\\/1\\" exact={[undefined]} strict={[undefined]} location={[undefined]}>
                          <Link to=\\"/1\\" className=\\"sc-htpNat jgryAF\\" style={[undefined]} aria-current={{...}} replace={false}>
                            <a className=\\"sc-htpNat jgryAF\\" style={[undefined]} aria-current={{...}} onClick={[Function]} href=\\"/1\\">
                              1
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                    </Styled(NavLink)>
                  </li>
                </ul>
              </styled.ul>
            </Styled(styled.ul)>
          </nav>
        </CategoryMenu>
      </Route>
    </withRouter(CategoryMenu)>
  </Router>
</MemoryRouter>"
`;
